### ISSUE TITLE
Cherry pick fix for handling clustered datastore name into 1.7 and 1.8

### ISSUE THREAD

**Issue by pshahzeb** [vmware/kubernetes#347]
Cherry pick fix for issue #312 (https://github.com/kubernetes/kubernetes/pull/54438) into prior releases -
- [ ] 1.7 release
- [x] 1.8 release

**Comment by divyenpatel** [comment]
Cherry Picked to 1.8: https://github.com/kubernetes/kubernetes/pull/54802

**Comment by pshahzeb** [comment]
This fix is not needed in release 1.7
This is due to a different logic of DiskIsAttached/DiskAreAttached logic.
In release 1.7 these method rely on `checkDiskAttached` -> `getVirtualDiskControllerKey` which calls `removeClusterFromVDiskPath` which handles clustered datastore name.

See the test output below for all 3 cases of clustered datastore test suite passing on release 1.7
```
From https://github.com/kubernetes/kubernetes
   6b03a43..cdbfff0  master     -> origin/master
   5e032cd..6653e44  release-1.9 -> origin/release-1.9
 * [new tag]         v1.10.0-alpha.1 -> v1.10.0-alpha.1
+ git checkout -b release-1.7 origin/release-1.7
Branch release-1.7 set up to track remote branch release-1.7 from origin.
Switched to a new branch 'release-1.7'
+ git branch
  master
* release-1.7
+ rm -rf _output/dockerized/bin/linux/amd64
+ mkdir -p _output/dockerized/bin/linux/amd64
+ cd _output/dockerized/bin/linux/amd64

....
...

-------------------------------[ Starting the Test ]-------------------------------
+ runTest '--ginkgo.focus=Volume\sProvisioning\sOn\sClustered\sDatastore'
+ GINKGO_FOCUS='--ginkgo.focus=Volume\sProvisioning\sOn\sClustered\sDatastore'
+ echo test '--ginkgo.focus=Volume\sProvisioning\sOn\sClustered\sDatastore'
test --ginkgo.focus=Volume\sProvisioning\sOn\sClustered\sDatastore
+ go run hack/e2e.go --check-version-skew=false --v --test '--test_args=--ginkgo.focus=Volume\sProvisioning\sOn\sClustered\sDatastore'
flag provided but not defined: -check-version-skew
Usage of /tmp/go-build859541669/command-line-arguments/_obj/exe/e2e:
  -get
    	go get -u kubetest if old or not installed (default true)
  -old duration
    	Consider kubetest old if it exceeds this (default 24h0m0s)
2017/12/18 21:45:12 e2e.go:54: NOTICE: go run hack/e2e.go is now a shim for test-infra/kubetest
2017/12/18 21:45:12 e2e.go:55:   Usage: go run hack/e2e.go [--get=true] [--old=24h0m0s] -- [KUBETEST_ARGS]
2017/12/18 21:45:12 e2e.go:56:   The separator is required to use --get or --old flags
2017/12/18 21:45:12 e2e.go:57:   The -- flag separator also suppresses this message
2017/12/18 21:45:12 e2e.go:76: Calling kubetest --check-version-skew=false --v --test --test_args=--ginkgo.focus=Volume\sProvisioning\sOn\sClustered\sDatastore...
2017/12/18 21:45:12 util.go:155: Running: ./cluster/kubectl.sh --match-server-version=false version
2017/12/18 21:45:12 util.go:157: Step './cluster/kubectl.sh --match-server-version=false version' finished in 313.504413ms
2017/12/18 21:45:12 util.go:155: Running: ./hack/e2e-internal/e2e-status.sh
Skeleton Provider: prepare-e2e not implemented
Client Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.0.12138+18dc58cc099fcc", GitCommit:"18dc58cc099fcced085b713840c67a9c2d2c22ff", GitTreeState:"clean", BuildDate:"2017-12-19T04:19:09Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.12-beta.0.46+40333bfae83bdf", GitCommit:"40333bfae83bdfe6245a2457a0c1a35c1c91b209", GitTreeState:"clean", BuildDate:"2017-12-15T12:35:58Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
2017/12/18 21:45:12 util.go:157: Step './hack/e2e-internal/e2e-status.sh' finished in 336.269166ms
2017/12/18 21:45:12 util.go:155: Running: ./hack/ginkgo-e2e.sh --ginkgo.focus=Volume\sProvisioning\sOn\sClustered\sDatastore
Conformance test: not doing test setup.
Dec 18 21:45:13.997: INFO: Overriding default scale value of zero to 1
Dec 18 21:45:13.998: INFO: Overriding default milliseconds value of zero to 5000
I1218 21:45:14.038287    5738 e2e.go:344] Starting e2e run "c8bc4628-e47f-11e7-aa7a-0242ac110002" on Ginkgo node 1
Running Suite: Kubernetes e2e suite
===================================
Random Seed: 1513662313 - Will randomize all specs
Will run 3 of 660 specs

Dec 18 21:45:14.074: INFO: >>> kubeConfig: /mnt/gitsource/kubernetes-anywhere-config/kubeconfig.json
Dec 18 21:45:14.078: INFO: Waiting up to 4h0m0s for all (but 0) nodes to be schedulable
Dec 18 21:45:14.262: INFO: Waiting up to 10m0s for all pods (need at least 0) in namespace 'kube-system' to be running and ready
Dec 18 21:45:14.409: INFO: 13 / 13 pods in namespace 'kube-system' are running and ready (0 seconds elapsed)
Dec 18 21:45:14.409: INFO: expected 4 pod replicas in namespace 'kube-system', 4 are Running and Ready.
Dec 18 21:45:14.445: INFO: Waiting for pods to enter Success, but no pods in "kube-system" match label map[name:e2e-image-puller]
Dec 18 21:45:14.445: INFO: Dumping network health container logs from all nodes...
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere] 
  verify dynamic provision with default parameter on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:123
[BeforeEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:130
STEP: Creating a kubernetes client
Dec 18 21:45:14.483: INFO: >>> kubeConfig: /mnt/gitsource/kubernetes-anywhere-config/kubeconfig.json
STEP: Building a namespace api object
STEP: Waiting for a default service account to be provisioned in namespace
[BeforeEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:57
[It] verify dynamic provision with default parameter on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:123
STEP: Creating Storage Class With storage policy params
STEP: Creating PVC using the Storage Class
STEP: Waiting for claim to be in bound phase
Dec 18 21:45:14.668: INFO: Waiting up to 5m0s for PersistentVolumeClaim pvc-j2hc1 to have phase Bound
Dec 18 21:45:14.704: INFO: PersistentVolumeClaim pvc-j2hc1 found but phase is Pending instead of Bound.
Dec 18 21:45:16.739: INFO: PersistentVolumeClaim pvc-j2hc1 found and phase=Bound (2.071284918s)
STEP: Creating pod to attach PV to the node
STEP: Verify the volume is accessible and available in the pod
Dec 18 21:45:23.169: INFO: [verifyVSphereDiskAttached] volumePath: "[dscl1/sharedVmfs-0] kubevols/kubernetes-dynamic-pvc-c9d5abee-e47f-11e7-93f7-005056a41f1b.vmdk", nodeName: "kubernetes-node3"
Dec 18 21:45:23.990: INFO: Running '/mnt/gitsource/e2e-1.7/kubernetes/_output/dockerized/bin/linux/amd64/kubectl --server=https://10.161.107.9 --kubeconfig=/mnt/gitsource/kubernetes-anywhere-config/kubeconfig.json exec pvc-tester-d9s0c --namespace=e2e-tests-volume-provision-5f978 -- /bin/touch /mnt/volume1/emptyFile.txt'
Dec 18 21:45:24.562: INFO: stderr: ""
Dec 18 21:45:24.562: INFO: stdout: ""
STEP: Deleting pod
Dec 18 21:45:24.562: INFO: Deleting pod pvc-tester-d9s0c
Dec 18 21:45:24.602: INFO: Waiting up to 5m0s for pod pvc-tester-d9s0c status to be terminated due to deadline exceeded
Dec 18 21:45:24.637: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (35.473355ms elapsed)
Dec 18 21:45:26.674: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (2.072047565s elapsed)
Dec 18 21:45:28.709: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (4.107189359s elapsed)
Dec 18 21:45:30.745: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (6.143116673s elapsed)
Dec 18 21:45:32.781: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (8.179097066s elapsed)
Dec 18 21:45:34.817: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (10.214642069s elapsed)
Dec 18 21:45:36.852: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (12.249802817s elapsed)
Dec 18 21:45:38.887: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (14.285536105s elapsed)
Dec 18 21:45:40.929: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (16.327469778s elapsed)
Dec 18 21:45:42.965: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (18.363510812s elapsed)
Dec 18 21:45:45.001: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (20.399334099s elapsed)
Dec 18 21:45:47.037: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (22.435401944s elapsed)
Dec 18 21:45:49.073: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (24.471244899s elapsed)
Dec 18 21:45:51.109: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (26.507138607s elapsed)
Dec 18 21:45:53.144: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (28.542572334s elapsed)
Dec 18 21:45:55.180: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: false) (30.578532795s elapsed)
Dec 18 21:45:57.216: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: false) (32.614150072s elapsed)
Dec 18 21:45:59.251: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: false) (34.649550394s elapsed)
Dec 18 21:46:01.287: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: false) (36.684773352s elapsed)
Dec 18 21:46:03.323: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: false) (38.721119856s elapsed)
Dec 18 21:46:05.359: INFO: Waiting for pod pvc-tester-d9s0c in namespace 'e2e-tests-volume-provision-5f978' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: false) (40.757045433s elapsed)
Dec 18 21:46:07.394: INFO: Pod "pvc-tester-d9s0c" in namespace "e2e-tests-volume-provision-5f978" disappeared. Error: pods "pvc-tester-d9s0c" not found
Dec 18 21:46:07.394: INFO: Ignore "not found" error above. Pod "pvc-tester-d9s0c" successfully deleted
STEP: Waiting for volumes to be detached from the node
Dec 18 21:46:17.394: INFO: [verifyVSphereDiskAttached] volumePath: "[dscl1/sharedVmfs-0] kubevols/kubernetes-dynamic-pvc-c9d5abee-e47f-11e7-93f7-005056a41f1b.vmdk", nodeName: "kubernetes-node3"
Dec 18 21:46:18.164: INFO: Volume "[dscl1/sharedVmfs-0] kubevols/kubernetes-dynamic-pvc-c9d5abee-e47f-11e7-93f7-005056a41f1b.vmdk" appears to have successfully detached from "kubernetes-node3".
Dec 18 21:46:18.164: INFO: Deleting PersistentVolumeClaim "pvc-j2hc1"
[AfterEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:131
Dec 18 21:46:18.238: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "e2e-tests-volume-provision-5f978" for this suite.
Dec 18 21:46:25.912: INFO: namespace: e2e-tests-volume-provision-5f978, resource: bindings, ignored listing per whitelist
Dec 18 21:46:26.086: INFO: namespace e2e-tests-volume-provision-5f978 deletion completed in 7.811269786s

• [SLOW TEST:71.603 seconds]
[k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:619
  verify dynamic provision with default parameter on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:123
------------------------------
SSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere] 
  verify static provisioning on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:113
[BeforeEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:130
STEP: Creating a kubernetes client
Dec 18 21:46:26.086: INFO: >>> kubeConfig: /mnt/gitsource/kubernetes-anywhere-config/kubeconfig.json
STEP: Building a namespace api object
STEP: Waiting for a default service account to be provisioned in namespace
[BeforeEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:57
[It] verify static provisioning on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:113
STEP: creating a test vsphere volume
STEP: Creating pod
STEP: Waiting for pod to be ready
STEP: Verifying volume is attached
Dec 18 21:46:46.946: INFO: [verifyVSphereDiskAttached] volumePath: "[dscl1/sharedVmfs-0] kubevols/e2e-vmdk-e2e-tests-volume-provision-xgnc4.vmdk", nodeName: "kubernetes-node1"
STEP: Deleting pod
Dec 18 21:46:47.612: INFO: Deleting pod vsphere-e2e-31nhj
Dec 18 21:46:47.651: INFO: Waiting up to 5m0s for pod vsphere-e2e-31nhj status to be terminated due to deadline exceeded
Dec 18 21:46:47.685: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (34.546866ms elapsed)
Dec 18 21:46:49.721: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (2.069951847s elapsed)
Dec 18 21:46:51.756: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (4.10508965s elapsed)
Dec 18 21:46:53.792: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (6.141286232s elapsed)
Dec 18 21:46:55.828: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (8.177360936s elapsed)
Dec 18 21:46:57.864: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (10.213051227s elapsed)
Dec 18 21:46:59.899: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (12.248670457s elapsed)
Dec 18 21:47:01.934: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (14.283698441s elapsed)
Dec 18 21:47:03.972: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (16.321014654s elapsed)
Dec 18 21:47:06.007: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (18.356521425s elapsed)
Dec 18 21:47:08.044: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (20.393226728s elapsed)
Dec 18 21:47:10.079: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (22.428812109s elapsed)
Dec 18 21:47:12.115: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (24.46427496s elapsed)
Dec 18 21:47:14.151: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (26.500815406s elapsed)
Dec 18 21:47:16.188: INFO: Waiting for pod vsphere-e2e-31nhj in namespace 'e2e-tests-volume-provision-xgnc4' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (28.536949693s elapsed)
STEP: Waiting for volumes to be detached from the node
Dec 18 21:47:28.224: INFO: [verifyVSphereDiskAttached] volumePath: "[dscl1/sharedVmfs-0] kubevols/e2e-vmdk-e2e-tests-volume-provision-xgnc4.vmdk", nodeName: "kubernetes-node1"
Dec 18 21:47:28.883: INFO: Volume "[dscl1/sharedVmfs-0] kubevols/e2e-vmdk-e2e-tests-volume-provision-xgnc4.vmdk" appears to have successfully detached from "kubernetes-node1".
STEP: Deleting the vsphere volume
[AfterEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:131
Dec 18 21:47:29.271: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "e2e-tests-volume-provision-xgnc4" for this suite.
Dec 18 21:47:36.186: INFO: namespace: e2e-tests-volume-provision-xgnc4, resource: bindings, ignored listing per whitelist
Dec 18 21:47:37.116: INFO: namespace e2e-tests-volume-provision-xgnc4 deletion completed in 7.808542803s

• [SLOW TEST:71.030 seconds]
[k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:619
  verify static provisioning on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:113
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
[k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere] 
  verify dynamic provision with spbm policy on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:135
[BeforeEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:130
STEP: Creating a kubernetes client
Dec 18 21:47:37.120: INFO: >>> kubeConfig: /mnt/gitsource/kubernetes-anywhere-config/kubeconfig.json
STEP: Building a namespace api object
STEP: Waiting for a default service account to be provisioned in namespace
[BeforeEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:57
[It] verify dynamic provision with spbm policy on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:135
STEP: Creating Storage Class With storage policy params
STEP: Creating PVC using the Storage Class
STEP: Waiting for claim to be in bound phase
Dec 18 21:47:37.305: INFO: Waiting up to 5m0s for PersistentVolumeClaim pvc-th849 to have phase Bound
Dec 18 21:47:37.340: INFO: PersistentVolumeClaim pvc-th849 found but phase is Pending instead of Bound.
Dec 18 21:47:39.376: INFO: PersistentVolumeClaim pvc-th849 found but phase is Pending instead of Bound.
Dec 18 21:47:41.411: INFO: PersistentVolumeClaim pvc-th849 found but phase is Pending instead of Bound.
Dec 18 21:47:43.446: INFO: PersistentVolumeClaim pvc-th849 found and phase=Bound (6.141138833s)
STEP: Creating pod to attach PV to the node
STEP: Verify the volume is accessible and available in the pod
Dec 18 21:47:55.851: INFO: [verifyVSphereDiskAttached] volumePath: "[sharedVmfs-0] kubevols/kubernetes-dynamic-pvc-1edaf0bd-e480-11e7-93f7-005056a41f1b.vmdk", nodeName: "kubernetes-node3"
Dec 18 21:47:56.642: INFO: Running '/mnt/gitsource/e2e-1.7/kubernetes/_output/dockerized/bin/linux/amd64/kubectl --server=https://10.161.107.9 --kubeconfig=/mnt/gitsource/kubernetes-anywhere-config/kubeconfig.json exec pvc-tester-trp8z --namespace=e2e-tests-volume-provision-svm8z -- /bin/touch /mnt/volume1/emptyFile.txt'
Dec 18 21:47:57.311: INFO: stderr: ""
Dec 18 21:47:57.311: INFO: stdout: ""
STEP: Deleting pod
Dec 18 21:47:57.312: INFO: Deleting pod pvc-tester-trp8z
Dec 18 21:47:57.349: INFO: Waiting up to 5m0s for pod pvc-tester-trp8z status to be terminated due to deadline exceeded
Dec 18 21:47:57.385: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (35.954037ms elapsed)
Dec 18 21:47:59.421: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (2.072105406s elapsed)
Dec 18 21:48:01.457: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (4.108044953s elapsed)
Dec 18 21:48:03.496: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (6.146476925s elapsed)
Dec 18 21:48:05.531: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (8.182102371s elapsed)
Dec 18 21:48:07.567: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (10.217883778s elapsed)
Dec 18 21:48:09.602: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (12.253125684s elapsed)
Dec 18 21:48:11.637: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (14.288258928s elapsed)
Dec 18 21:48:13.673: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (16.323932442s elapsed)
Dec 18 21:48:15.709: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (18.359514319s elapsed)
Dec 18 21:48:17.744: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (20.394759688s elapsed)
Dec 18 21:48:19.779: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (22.430131897s elapsed)
Dec 18 21:48:21.815: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (24.465932953s elapsed)
Dec 18 21:48:23.850: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (26.501162391s elapsed)
Dec 18 21:48:25.886: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: true) (28.536443398s elapsed)
Dec 18 21:48:27.922: INFO: Waiting for pod pvc-tester-trp8z in namespace 'e2e-tests-volume-provision-svm8z' status to be 'terminated due to deadline exceeded'(found phase: "Running", readiness: false) (30.572804602s elapsed)
Dec 18 21:48:29.957: INFO: Pod "pvc-tester-trp8z" in namespace "e2e-tests-volume-provision-svm8z" disappeared. Error: pods "pvc-tester-trp8z" not found
Dec 18 21:48:29.957: INFO: Ignore "not found" error above. Pod "pvc-tester-trp8z" successfully deleted
STEP: Waiting for volumes to be detached from the node
Dec 18 21:48:39.957: INFO: [verifyVSphereDiskAttached] volumePath: "[sharedVmfs-0] kubevols/kubernetes-dynamic-pvc-1edaf0bd-e480-11e7-93f7-005056a41f1b.vmdk", nodeName: "kubernetes-node3"
Dec 18 21:48:40.756: INFO: Volume "[sharedVmfs-0] kubevols/kubernetes-dynamic-pvc-1edaf0bd-e480-11e7-93f7-005056a41f1b.vmdk" appears to have successfully detached from "kubernetes-node3".
Dec 18 21:48:40.757: INFO: Deleting PersistentVolumeClaim "pvc-th849"
[AfterEach] [k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:131
Dec 18 21:48:40.835: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "e2e-tests-volume-provision-svm8z" for this suite.
Dec 18 21:48:48.633: INFO: namespace: e2e-tests-volume-provision-svm8z, resource: bindings, ignored listing per whitelist
Dec 18 21:48:48.703: INFO: namespace e2e-tests-volume-provision-svm8z deletion completed in 7.830179418s

• [SLOW TEST:71.583 seconds]
[k8s.io] Volume Provisioning On Clustered Datastore [Feature:vsphere]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:619
  verify dynamic provision with spbm policy on clustered datastore
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/vsphere_volume_cluster_ds.go:135
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSDec 18 21:48:48.710: INFO: Running AfterSuite actions on all node
Dec 18 21:48:48.710: INFO: Running AfterSuite actions on node 1

Ran 3 of 660 Specs in 214.636 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 657 Skipped PASS
```
